Source/Scene/Batched3DModel3DTileContent.js

```diff
@@ -199,25 +199,49 @@ define([
         var byteLength = view.getUint32(byteOffset, true);
         byteOffset += sizeOfUint32;
 
+        var batchTableJsonByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var batchTableBinaryByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
         var batchLength = view.getUint32(byteOffset, true);
-        this._featuresLength = batchLength;
         byteOffset += sizeOfUint32;
 
+        // TODO : remove this legacy check before merging into master
+        // Legacy header:  [batchLength] [batchTableByteLength]
+        // Current header: [batchTableJsonByteLength] [batchTableBinaryByteLength] [batchLength]
+        // If the header is in the legacy format 'batchLength' will be the start of the JSON string (a quotation mark) or the glTF magic.
+        // Accordingly the first byte of uint32 will be either 0x22 or 0x67 and so the uint32 will exceed any reasonable 'batchLength'.
+        if (batchLength > 10000000) {
+            byteOffset -= sizeOfUint32;
+            batchLength = batchTableJsonByteLength;
+            batchTableJsonByteLength = batchTableBinaryByteLength;
+            batchTableBinaryByteLength = 0;
+            console.log('Warning: b3dm header is using the legacy format [batchLength] [batchTableByteLength]. Please update to the latest b3dm format: https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/TileFormats/Batched3DModel/README.md.');
+        }
+
+        this._featuresLength = batchLength;
         var batchTableResources = new Cesium3DTileBatchTableResources(this, batchLength);
         this.batchTableResources = batchTableResources;
 
-        var batchTableByteLength = view.getUint32(byteOffset, true);
-        byteOffset += sizeOfUint32;
-        if (batchTableByteLength > 0) {
-            var batchTableString = getStringFromTypedArray(uint8Array, byteOffset, batchTableByteLength);
-            byteOffset += batchTableByteLength;
-
+        if (batchTableJsonByteLength > 0) {
             // PERFORMANCE_IDEA: is it possible to allocate this on-demand?  Perhaps keep the
             // arraybuffer/string compressed in memory and then decompress it when it is first accessed.
             //
             // We could also make another request for it, but that would make the property set/get
             // API async, and would double the number of numbers in some cases.
-            batchTableResources.batchTable = JSON.parse(batchTableString);
+            var batchTableString = getStringFromTypedArray(uint8Array, byteOffset, batchTableJsonByteLength);
+            var batchTableJson = JSON.parse(batchTableString);
+            byteOffset += batchTableJsonByteLength;
+
+            var batchTableBinary;
+            if (batchTableBinaryByteLength > 0) {
+                // Has a batch table binary
+                batchTableBinary = new Uint8Array(arrayBuffer, byteOffset, batchTableBinaryByteLength);
```

Comment:
I don't know how the GC works, but I wonder if holding on to this subset of the original `arrayBuffer` forces the GC to keep the entire `arrayBuffer` in memory.  @shunter do you know?